restore_drosera.sh

```
@@ -506,37 +506,138 @@ source /root/.bashrc 2>/dev/null || true
 print_color $YELLOW "Установка зависимостей трапа..."
 /root/.bun/bin/bun install >/dev/null 2>&1
 /root/.foundry/bin/forge build >/dev/null 2>&1
-check_status "Компиляция трапа"
+check_status "Компиляция трапа" # This initial build compiles template contracts
 
-# Создание конфигурации drosera.toml
-print_color $YELLOW "Настройка файла конфигурации drosera.toml..."
+# Конфигурация и применение трапа в зависимости от SETUP_CADET
+if [ "$SETUP_CADET" = true ]; then
+    print_header "НАСТРОЙКА CADET РОЛИ (ТРАП)" # Modified header to be more specific
+    cd ~/my-drosera-trap # Ensure correct directory
```

Shellcheck (0.10.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)